docs/databases/migrations.md

```diff
@@ -4,4 +4,568 @@ title: GraphQL Migrations
+primary key for the `Note` model and a custom primary key for the `Comment` model. We will see how these looks like in details 
+for each model below:
+
+```sql
+\d comment
+              Table "public.comment"
+   Column    |          Type          | Modifiers 
+-------------+------------------------+-----------
+ text        | character varying(255) | not null
+ description | character varying(255) | 
+ noteId      | integer                | 
+Indexes:
+    "comment_pkey" PRIMARY KEY, btree (text)
+Foreign-key constraints:
+    "comment_noteid_foreign" FOREIGN KEY ("noteId") REFERENCES note(id)
+```
+
+Table details for the `Commet` type shows that we have a `text` as the primary key and `noteId` as a foreign key on the `Note` type because of the
+relationship we defined. 
+
+Here is how the `Note` table looks like:
+
+```sql
+\d note
+                                   Table "public.note"
+   Column    |          Type          |                     Modifiers                     
+-------------+------------------------+---------------------------------------------------
+ title       | character varying(255) | not null
+ description | character varying(255) | 
+ id          | integer                | not null default nextval('note_id_seq'::regclass)
+Indexes:
+    "note_pkey" PRIMARY KEY, btree (id)
+Referenced by:
+    TABLE "comment" CONSTRAINT "comment_noteid_foreign" FOREIGN KEY ("noteId") REFERENCES note(id)
+```
+
+Graphback migrations created the auto-incremented primary key by inferring the usage of `id: ID!` field on the `Note` model.
+Additionally we can see that this primary key is referenced by the `comment` table because of the relationship.
+
+
+#### Default field value
+
+You can specify a default value using the `@default` field annotation as shown below.
+
+```graphql
+type Note {
+  id: ID!
+  title: String!
+  """
+  Define a default value
+  @default(value: false)
+  """
+  complete: Boolean
+}
+```
+
+#### Custom column type
+
+You can specify a custom type usig the `@db` field annotation as shown below:
+
+```graphql
+type Note {
+  id: ID!
+  """
+  Define a custom json type
+  @db(type: 'json')
+  """
+  comments: [Comment]
+}
+
+type Comment {
+  text: String
+  description: String
+}
+```
+
+#### Changing column length in Graphback
+
+By default string columns will be created as `varchar` with a column length of `255`. 
+This can be configured to any length using the `@db` field annnotation and the `length` argument:
+
+```graphql
+type Note {
+  id: ID!
+  """
+  Define custom column length
+  @db(length: 100)
+  """
+  title: String!
+}
+```
+
+#### Ignore a field 
+
+Sometimes our business model can contain more fields and some of these fields may only be there for representation purposes.
+We can safely ignore generating columns for these fields using the `@db(skip: true)` annotation on the corresponding field. 
+
+```graphql
+type Note {
+  id: ID!
+  """
+  Define custom column length
+  @db(skip: true)
```

Comment:
Though GraphQL Migrations supports this, Graphback does not recognise it (in that it will still create input fields.

Reply:
Yeap, I was only focused on the migration part. 

I guess it is worth to add a note. WDYT?

Reply:
I think so yeah.